src/battle_script_commands.c

```diff
@@ -4553,20 +4550,27 @@ static void Cmd_moveend(void)
                 gBattlerTarget = gActiveBattler;
                 gHitMarker &= ~(HITMARKER_SWAP_ATTACKER_TARGET);
             }
-            if (gHitMarker & HITMARKER_ATTACKSTRING_PRINTED)
```

Comment:
Could you bring back this check?

Reply:
The check is there, just a few lines down - it makes sure that "dancerUsedMove" is 0 before setting the last used move value.

Reply:
The check is there on line 4557 - first is checks that the last move used wasn't due to Dancer.